R/rowAvgsPerColSet.R

```diff
@@ -128,6 +128,9 @@ rowAvgsPerColSet <- function(X, W = NULL, rows = NULL, S,
     Zjj
   })
 
+  # Ensure that Z is a matrix and not a vector
+  Z <- as.matrix(Z)
```

Comment:
`stopifnot(dimX[1] == 1L)` will break if `nrow(X) == 0` (caught by tests in `tests/rowAvgsPerColSet_subset.R`) whereas `as.matrix(Z)` passes the tests

Reply:
Ah... (of course) so the assertion should be `stopifnot(dimX[1] <= 1L)` then?  The reason why I'd like to avoid `as.matrix()` is that it is "too magic" and might fix other things as well - things that are bug and that should generate an error.